### PR TITLE
Suppress user and password info output via Config structure.

### LINKED ISSUE #935: [security] mask/remove the user and password for config debug
**Summary**

```
[2021-06-28T03:13:49Z INFO  fuse_query] Config { log_level: "INFO", log_dir: "./_logs", num_cpus: 16, mysql_handler_host: "127.0.0.1", mysql_handler_port: 3307, mysql_handler_thread_num: 256, clickhouse_handler_host: "127.0.0.1", clickhouse_handler_port: 9000, clickhouse_handler_thread_num: 256, flight_api_address: "127.0.0.1:9090", http_api_address: "127.0.0.1:8080", metric_api_address: "127.0.0.1:7070", store_api_address: "127.0.0.1:9191", store_api_username: "root", store_api_password: "root", config_file: "" }
```

Remove user/password config from debug.

### PR DESCRIPTION
**Pull request by wangzhen11aaa**
I hereby agree to the terms of the CLA available at: https://datafuse.rs/policies/cla/

## Summary

Draft pull request. May not the best solution, Please give advice about the improvement.

## Changelog

Rewrite the Debug trait for the Config structure. This fix will affect the extension of the Config, and seems not the best answer.

- Improvement

- Not for changelog (changelog entry is not required)

## Related Issues
fixes #935 

## Test Plan

Unit Tests
Stateless Tests

### CODE REVIEW

File: fusequery/query/src/configs/config.rs [hunk 1]
```diff
@@ -27,8 +29,19 @@ lazy_static! {
...
+        write!(f, "{{ log_level: \"{}\", log_dir: \"{}\", num_cpus: {}, mysql_handler_host: \"{}\",
+        mysql_handler_port: {}, mysql_handler_thread_num: {}, clickhouse_handler_host: \"{}\", clickhouse_handler_clickhouse_handler_port: {}, 
+        clickhouse_handler_thread_num: {}, flight_api_address: \"{}\", http_api_address: \"{}\", metric_api_address: \"{}\", 
+        store_api_address: \"{}\", config_file: \"{}\", }}",
+        self.log_level, self.log_dir, self.num_cpus, self.mysql_handler_host, self.mysql_handler_port,
+        self.mysql_handler_thread_num, self.clickhouse_handler_host, self.clickhouse_handler_port,
+        self.clickhouse_handler_thread_num, self.flight_api_address, self.http_api_address, self.metric_api_address,
+        self.store_api_address, self.config_file)
+    }
+}
```
Comment (zhang2014) [inline fusequery/query/src/configs/config.rs:43]: maybe should add new line : )

File: fusequery/query/src/configs/config.rs [hunk 2]
```diff
@@ -27,8 +29,19 @@ lazy_static! {
         ver
     };
 }
-
-#[derive(Clone, Debug, serde::Deserialize, PartialEq, StructOpt, StructOptToml)]
+impl fmt::Debug for Config{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{{ log_level: \"{}\", log_dir: \"{}\", num_cpus: {}, mysql_handler_host: \"{}\",
```
Comment (zhang2014) [inline fusequery/query/src/configs/config.rs:34]: It looks like a JSON. maybe we can consider the following two implementations:
1. Reference https://github.com/rust-lang/rust/blob/master/library/core/src/fmt/mod.rs#L523 to implemented.
2. Use serde::Serialize and [skip field attribute](https://serde.rs/field-attrs.html#skip)

Comment (wangzhen11aaa) [reply fusequery/query/src/configs/config.rs:34]: I will follow up. Thx.

Comment (zhang2014) [reply fusequery/query/src/configs/config.rs:34]: Or we can implement the following code and change the type of store_api_password type to Password
``` rust
struct Password {
    plain_text : String
}

impl Debug for Password {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        write!(f, "******");
    }
}
```

Comment (wangzhen11aaa) [reply fusequery/query/src/configs/config.rs:34]: .

**Comment by wangzhen11aaa** [comment]
Please review and give more precious advice.

**Comment by wangzhen11aaa** [comment]
https://doc.rust-lang.org/std/string/trait.ToString.html
We need implement to_string() function to work with env_helper! macro.

File: fusequery/query/src/configs/config.rs [hunk 3]
```diff
@@ -135,16 +137,68 @@ pub struct Config {
...
+    #[structopt(long, env = "STORE_API_PASSWORD", default_value = "root")]
+    pub store_api_password: Password,
 
-    #[structopt(long, env = STORE_API_PASSWORD, default_value = "root")]
-    pub store_api_password: String,
 
     #[structopt(long, short = "c", env = CONFIG_FILE, default_value = "")]
     pub config_file: String,
 }
+#[derive(Clone, serde::Deserialize, PartialEq, StructOpt, StructOptToml)]
```
Comment (zhang2014) [inline fusequery/query/src/configs/config.rs:151]: need new line.

File: fusequery/query/src/configs/config.rs [hunk 4]
```diff
@@ -135,16 +137,68 @@ pub struct Config {
...
+        })
+    }
+}
+
+impl fmt::Debug for Password{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "******")
+    }
+}
+#[derive(Clone, serde::Deserialize, PartialEq, StructOpt, StructOptToml)]
```
Comment (zhang2014) [inline fusequery/query/src/configs/config.rs:177]: need new line

File: fusequery/query/src/configs/config.rs [hunk 5]
```diff
@@ -135,16 +137,68 @@ pub struct Config {
...
+    pub store_api_password: Password,
 
-    #[structopt(long, env = STORE_API_PASSWORD, default_value = "root")]
-    pub store_api_password: String,
 
     #[structopt(long, short = "c", env = CONFIG_FILE, default_value = "")]
     pub config_file: String,
 }
+#[derive(Clone, serde::Deserialize, PartialEq, StructOpt, StructOptToml)]
+#[serde(default)]
```
Comment (zhang2014) [inline fusequery/query/src/configs/config.rs:152]: Why we need this?

File: fusequery/query/src/configs/config.rs (hunk 3 above)
Comment (wangzhen11aaa) [reply fusequery/query/src/configs/config.rs:151]: ok

File: fusequery/query/src/configs/config.rs (hunk 4 above)
Comment (wangzhen11aaa) [reply fusequery/query/src/configs/config.rs:177]: Do we have some tool or extension (in the Vscode) can do this formatting thing automatically? Give a suggestion if have. Thanks.

File: fusequery/query/src/configs/config.rs (hunk 5 above)
Comment (wangzhen11aaa) [reply fusequery/query/src/configs/config.rs:152]: I thought this self-defined structure may need the same traits as Config did.

File: fusequery/query/src/configs/config.rs (hunk 4 above)
Comment (wangzhen11aaa) [reply fusequery/query/src/configs/config.rs:177]: image
This Rust Assist extesion looks good. Do we need this format tool？

Comment (zhang2014) [reply fusequery/query/src/configs/config.rs:177]: We use rust-fmt to format the code. You can use `make lint` to check the format

File: fusequery/query/src/configs/config.rs (hunk 5 above)
Comment (zhang2014) [reply fusequery/query/src/configs/config.rs:152]: Understand.

**Comment by wangzhen11aaa** [comment]
> We need implement to_string() function to work with env_helper! macro.
> May be like this.
> https://doc.rust-lang.org/std/string/trait.ToString.html

Need to implement ToString trait for User and Password.

File: fusequery/query/src/configs/config.rs [hunk 6]
```diff
@@ -136,16 +139,82 @@ pub struct Config {
     #[structopt(long, env = STORE_API_ADDRESS, default_value = "127.0.0.1:9191")]
     pub store_api_address: String,
 
-    #[structopt(long, env = STORE_API_USERNAME, default_value = "root")]
-    pub store_api_username: String,
+    #[structopt(long, env = "STORE_API_USERNAME", default_value = "root")]
```
Comment (zhang2014) [inline fusequery/query/src/configs/config.rs:142]: Why need change STORE_API_USERNAME to "STORE_API_USERNAME"?

**Review by zhang2014 — APPROVED** [review]
LGTM
Thank you @wangzhen11aaa

**Pull request merged**